### ISSUE TITLE
Using GQL cursors with Lockbox causes unnecessary decryption

### ISSUE THREAD

**Issue by benjaminoakes** [rmosolgo/graphql-ruby#3149]
There is an interaction with [Lockbox](https://github.com/ankane/lockbox) that causes more decryption to occur than is necessary when GraphQL Pro cursors are used.

(I'm also using [KMS Encrypted](https://github.com/ankane/kms_encrypted), but that only seems relevant because the extra decryption would mean more calls to a KMS service and those extra calls cost money.)

If I have a model that has a Lockbox encrypted column that has a non-nil value, [`Decrypt Data Key` events will be logged by KMS Encrypted](https://github.com/ankane/kms_encrypted/blob/v1.2.0/lib/kms_encrypted/log_subscriber.rb#L7) even though the GQL query does not load the encrypted data.  I dug into why the decryption happens, and it only occurs when cursors (e.g., `cursor`, `startCursor`, `endCursor`) are a part of the GQL query.  Take the cursors out of the GQL query, and no unexpected decryption happens.

I patched the decryption to obtain a stack trace to discover why this happens.  I noticed that GraphQL Pro calls the `attributes` method on my model to access `cursor_0`, `cursor_1`, etc, which come from a SQL `SELECT`.  Unfortunately, [the `attributes` method is patched by Lockbox to provide decrypted values](https://github.com/ankane/lockbox/blob/v0.4.7/lib/lockbox/model.rb#L85-L99).  This is the cause of the extra decryption because GraphQL Pro is requesting more data than is necessary by using `attributes`.  Also, to emphasize: improving this by loading less information would help all users of GraphQL Pro.

There are likely several solutions to this, but I would suggest accessing `cursor_N` as a method instead.

In pseudocode, this is what I'm describing:

```
>> User.select("*, 1 as one").first.attributes[:one]
Decrypt Data Key (0.2ms)  Context: {:model_name=>"User", :model_id=>2}
=> 1
>> User.select("*, 1 as one").first.respond_to?(:one)
=> true
>> User.select("*, 1 as one").first.public_send(:one)
=> 1
```

The call to `attributes` triggers `Decrypt Data Key`, but the method version does not.

**Gem versions, for posterity:**

  - GraphQL v1.10.14
  - GraphQL Pro v1.11.0
  - KMS Encrypted v1.2.0
  - Lockbox v0.4.7
  - Rails v5.2.4.3

**Comment by rmosolgo** [comment]
Hey, sorry for the hangup and thanks for the great writeup. 

> accessing `cursor_N` as a method instead.

That's a great suggestion, I'll look into it. I suspect that the `.attributes[...]` lookup is an artifact of history: in an old version, instead of aliasing the fields, I tried to access them directly, but I had to work around any other methods defined on the model. Now that the fields are aliased, we could reliably use those methods to retrieve their values. 

I'll give it a shot and report back here.
